src/kmrbase.c

```diff
@@ -122,6 +134,11 @@ kmr_init_ff(int kf, struct kmr_option opt, struct kmr_file_option fopt)
 int
 kmr_fin(void)
 {
+#if KMR_TRACE_ENABLE
```

Comment:
Calling these three functions in kmr_free_context is better, when the tracer is implemented as an option to kmr struct.